### ISSUE TITLE
SQL error logged while importing from Dropbox

### ISSUE THREAD

**Issue by alexolivan** [ElvishArtisan/rivendell#899]
Hi!

I'm importing some songs in my testing install (latest commit as of 17 Aug 2023)
And, looking at syslog, I'm quite confident I see a consistent behavior, where for each file ingested, an SQL error is thrown into the log... here's  how it looks like:

```
Aug 17 13:03:21 tgnradio dropbox[37]: Importing file "Breakfast Club - Right On Track.mp3" [right on track] to cart 011026 ...
Aug 17 13:03:22 tgnradio ripcd: added new connection 22
Aug 17 13:03:22 tgnradio rdxport.cgi: processing RDXPORT_COMMAND_IMPORT
Aug 17 13:03:34 tgnradio ripcd: sent notification: "NOTIFY CART MODIFY 11026" to 239.192.255.72:20539
Aug 17 13:03:34 tgnradio ripcd: closed connection 22
Aug 17 13:03:34 tgnradio dropbox[37]: done.
Aug 17 13:03:34 tgnradio ripcd: cleaned up connection 22
Aug 17 13:03:34 tgnradio ripcd: sent notification: "NOTIFY CART ADD 11026" to 239.192.255.72:20539
Aug 17 13:03:34 tgnradio dropbox[37]: Deleted file "Breakfast Club - Right On Track.mp3"
Aug 17 13:03:34 tgnradio rdlibrary: DB connection re-established
Aug 17 13:03:34 tgnradio rdlibrary: invalid SQL or failed DB connection[Unknown column 'CART_SCHED_CODES.SCHED_CODE' in 'where clause' QMYSQL: Unable to execute query]: select `CART`.`NUMBER` from `CART` left join `GROUPS` on `CART`.`GROUP_NAME`=`GROUPS`.`NAME` left join `CUTS` on `CART`.`NUMBER`=`CUTS`.`CART_NUMBER`  where (`CART`.`NUMBER`=11026) && ((`CART`.`TYPE`=1)||(`CART`.`TYPE`=2)||(`CART`.`TYPE`=3)) &&  ((`CART`.`GROUP_NAME`="BEDS")||(`CART`.`GROUP_NAME`="IDENTS")||(`CART`.`GROUP_NAME`="LEGAL")||(`CART`.`GROUP_NAME`="LOGOS")||(`CART`.`GROUP_NAME`="MACROS")||(`CART`.`GROUP_NAME`="MUSIC")||(`CART`.`GROUP_NAME`="RECORDS")||(`CART`.`GROUP_NAME`="STINGS")||(`CART`.`GROUP_NAME`="TEMP")||(`CART`.`GROUP_NAME`="TEST")||(`CART`.`GROUP_NAME`="TRAFFIC")||(`CART`.`GROUP_NAME`="VOICETRACK"))&&(`CART_SCHED_CODES`.`SCHED_CODE`='Ingested') 
```

Hope it helps spotting something amiss. Apparently, the importing does actually work (cart is created with a valid audio cut)

Cheers.

**Comment by ElvishArtisan** [comment]
Pretty sure that this is already fixed in current Git 'v4'. What version of Rivendell is this?

**Comment by alexolivan** [comment]
Hi fred.

Just apt-upgraded after last build... so bleeding edge snapshot:
```
root@tgnradio:~# apt-cache policy rivendell
rivendell:
  Installed: 4.1.0-1+git20230816.faaf780
  Candidate: 4.1.0-1+git20230816.faaf780
  Version table:
 *** 4.1.0-1+git20230816.faaf780 500
        500 https://repo.alexolivan.com/debian bullseye/rivendell/snapshots/main amd64 Packages
        100 /var/lib/dpkg/status
```

EDIT:
I've ingested some 150 files on a dropbox folder so I can confirm this happens each and every time a file is processed

**Comment by ElvishArtisan** [comment]
'Kay, thanks. I will poke at this some more...

**Comment by alexolivan** [comment]
Hi again.

Looking backwards on the logs and analyzing what was going on, I realize that maybe all this has something to do with the fact that RDLibrary module was open while the files were being ingested...
Moreover, looking closely at the SQL statement, I recognize in the WHERE filtering the scheduler code 'Ingested' (which I have configured on my Dropboxes in order to be able to find new media on the library), and turns out that that sched code was actually selected on the dropdown menu of the GUI module to filter carts.

So maybe the way to reproduce this is to ingest a new file, on a dropbox that ads certain sched code, while having RDLibrary open and with that sched code selected as filter.

So far, the behavior or RDlibrary was quite nice during the whole ingest process, updating the view as new carts were added, so, apparently, the front-end was working flawlessly... but maybe the issue does not come from backend but rather from front-end after all... not sure... but maybe.

Cheers.

**Comment by ElvishArtisan** [comment]
> So maybe the way to reproduce this is to ingest a new file, on a dropbox that ads certain sched code, while having RDLibrary open and with that sched code selected as filter.

Yup. That goosed it!

**Comment by ElvishArtisan** [comment]
Fixed in 6c1abab. Please test!

**Comment by alexolivan** [comment]
OK... closing this :-)

Cheers!
